yarn-project/aztec.js/src/contract/deploy_method.ts

```diff
@@ -58,9 +58,10 @@ export class DeployMethod<TContract extends ContractBase = Contract> extends Bas
     private artifact: ContractArtifact,
     private postDeployCtor: (address: AztecAddress, wallet: Wallet) => Promise<TContract>,
     private args: any[] = [],
+    constructorName: string = 'constructor',
   ) {
     super(wallet);
-    const constructorArtifact = artifact.functions.find(f => f.name === 'constructor');
+    const constructorArtifact = artifact.functions.find(f => f.name === constructorName);
     if (!constructorArtifact) {
       throw new Error('Cannot find constructor in the artifact.');
```

Comment:
Nit: probably helps to include the variable name in the error here.

Reply:
Will do in a subsequent PR!